central/processlisteningonport/service/service_impl.go

```diff
@@ -48,12 +49,17 @@ func (s *serviceImpl) GetListeningEndpoints(
 	req *v1.GetProcessesListeningOnPortsRequest,
 ) (*v1.GetProcessesListeningOnPortsResponse, error) {
 	deployment := req.GetDeploymentId()
+	page := req.GetPagination()
 	processesListeningOnPorts, err := s.dataStore.GetProcessListeningOnPort(ctx, deployment)
```

Comment:
```suggestion
	processesListeningOnPorts, err := s.dataStore.GetProcessListeningOnPort(ctx, deployment, page)
```

> The pagination is done in central rather than by the query, because central does some filtering so using LIMIT in the query does not guarantee the number of listening endpoints in the response.

What filering do you have in mind. From code it looks like it's just a SQL query that could be augumented with `LIMIT` and `OFFSET` although it's not the best approach (maybe we should introcude last seen here? cc. @erthalion 

https://github.com/stackrox/stackrox/blob/2ba0186029091057dbaf2a26d1b1d731e55fb6f7/central/processlisteningonport/store/postgres/full_store.go#L39-L44

Reply:
There is a filter here https://github.com/stackrox/stackrox/blob/1c34180fb74474c0021335fa1f49d0391ae26315/central/processlisteningonport/store/postgres/full_store.go#L227

It is not really possible to do the filter in the query, because it involves data from the serialized column.

Reply:
OK, but we should sort the results somehow to guarantee the pagination will work properly as it's not guaranteed the order will be the same next time we call the query

Reply:
keep them separate